src/cli_driver.rs

```diff
@@ -449,14 +448,30 @@ impl ProcessingSession {
             };
 
             if let Some(par) = tex_path.parent() {
-                fs_root = par;
+                output_path = par;
                 io_builder.filesystem_root(par);
             } else {
                 return Err(ErrorKind::Msg(format!("can't figure out a parent directory for input path \"{}\"",
                                                   tex_path.to_string_lossy())).into());
             }
         }
 
+        if let Some(dir) = args.value_of("outdir") {
+            output_path = {
+                let tmp = Path::new(dir);
+                match tmp.extension() {
```

Comment:
I don't think this code should special-case the circumstance where the argument path has an extension. It's not common but sometimes directory names have extensions like files, and in those cases you'll get surprising behavior because the program will alter the path that you pass it.